### PR TITLE
Fix: 96 Reconcile As Env

### LINKED ISSUE #96: Reconcile Loop time as ENV
- Can we use an ENV for reconcile loop time, we can keep 10 sec to default
- https://github.com/druid-io/druid-operator/blob/master/pkg/controller/druid/druid_controller.go#L103 
- Their are cases where we want to lower the reconcile time etc, its more flexible in an env :)

### PR DESCRIPTION
**Pull request by AdheipSingh**
Fixes #96 

Need to add this in docs 

### Description

<hr>

This PR has:
- [x] been tested on a real K8S cluster to ensure creation of a brand new Druid cluster works.
- [x] been tested for backward compatibility on a real K*S cluster by applying the changes introduced here on an existing Druid cluster. If there are any backward incompatible changes then they have been noted in the PR description.
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added documentation for new or modified features or behaviors.

<hr>

##### Key changed/added files in this PR
 * `druid_controller.go`
 * `operator.yaml`

### CODE REVIEW

File: pkg/controller/druid/druid_controller.go [hunk 1]
```diff
@@ -100,6 +101,22 @@ func (r *ReconcileDruid) Reconcile(request reconcile.Request) (reconcile.Result,
...
 	} else {
-		return reconcile.Result{RequeueAfter: time.Second * 10}, nil
+		return reconcile.Result{RequeueAfter: lookupReconcileTime()}, nil
+	}
+}
+
+// lookupReconcileTime shall return reconcile time if set as ENV
+// else defaults to 10s
+func lookupReconcileTime() time.Duration {
+	val, exists :=  os.LookupEnv("RECONCILE_TIME")
```
Comment (himanshug) [inline pkg/controller/druid/druid_controller.go:111]: nit: RECONCILE_WAIT_ON_ERROR ?


also, can we parse that on startup and fail the operator startup if env variable was provided but invalid?

Comment (AdheipSingh) [reply pkg/controller/druid/druid_controller.go:111]: sure make sense
But why WAIT ON ERROR ? curious :thinking:

**Comment by AdheipSingh** [comment]
since reconcile method attached to reconcileDruid struct and this will be initalized in main.go, i added this field to the struct itself, so it'll be called automatically during start on controller.
Plus i feel its better to hook such fields to the struct itself for better understanding/readability.
Have tested this in dev, seems fine. 

Once approved ill add a few lines in the doc @himanshug

File: pkg/controller/druid/druid_controller.go (hunk 1 above)
Comment (himanshug) [reply pkg/controller/druid/druid_controller.go]: I re-read this code a bit and realized that env is actually used for non-error case, `RECONCILE_WAIT` was the correct term, my mistake.

Comment (AdheipSingh) [reply pkg/controller/druid/druid_controller.go]: ah ha i have been using that in my local :), will send the change and doc too. @himanshug

**Pull request merged**